### PR TITLE
feature/upgrade-all-packages-15-05-2023

### PR DESCRIPTION
**Pull request by alexndlm**

### CODE REVIEW

File: src/Rector/InheritDocRector.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ public function refactor(Node $node): ?Node
         $children = $phpDocInfo->getPhpDocNode()->children;
 
         foreach ($children as $key => $child) {
-            if (\strval($child->getAttribute('orig_node')) === self::INHERITDOC_INCORRECT_ANNOTATION) {
+            if ($child->getAttribute('orig_node') . '' === self::INHERITDOC_INCORRECT_ANNOTATION) {
```
Comment (alexndlm) [inline src/Rector/InheritDocRector.php:72]: Due to https://github.com/phpstan/phpstan/issues/9295

File: tests/Sniffs/Namespaces/Psr4Sniff/Psr4SniffTest.php [hunk 2]
```diff
@@ -4,36 +4,45 @@
...
+        ];
     }
 
     /**
-     * @dataProvider providerTestSniff
+     * @param array<int, array{line: int, code: string}>|null $expectedErrors
+     *
+     * @dataProvider provideFixtures
      */
-    public function testSniff(SmartFileInfo $smartFileInfo, int $expectedErrorCount): void
```
Comment (alexndlm) [inline tests/Sniffs/Namespaces/Psr4Sniff/Psr4SniffTest.php:30]: This test was not worked before. Due to specific sniff implementation, we need a special test.

File: ecs.php [hunk 3]
```diff
@@ -33,7 +33,7 @@
     $ecsConfig->skip([
         'tests/*/Fixture/*',
         AlphabeticallySortedArrayKeysSniff::class => [
-            'tests/Output/PrinterTest.php',
+            'tests/*',
```
Comment (roman-eonx) [inline ecs.php:36]: Why do we want to ignore the whole dir?

File: src/Rector/InheritDocRector.php (hunk 1 above)
Comment (roman-eonx) [reply src/Rector/InheritDocRector.php:72]: I don't really like the current solution. Do we have other options? Does casting to string work?

**Review by roman-eonx — CHANGES_REQUESTED** [review]

File: ecs.php (hunk 3 above)
Comment (alexndlm) [reply ecs.php:36]: Primarily due to `dataProvider`. We do this for all other projects.

File: src/Rector/InheritDocRector.php (hunk 1 above)
Comment (alexndlm) [reply src/Rector/InheritDocRector.php:72]: No :( 
Casting and `strval` do not work.

Comment (roman-eonx) [reply src/Rector/InheritDocRector.php:72]: Let's discuss this with the team then. Writing `. ''` and `* 1` everywhere will not make our code easier to read.

Comment (alexndlm) [reply src/Rector/InheritDocRector.php]: I have refactored. Please check.

**Review by roman-eonx — APPROVED** [review]

File: tests/Sniffs/AbstractSniffTestCase.php [hunk 4]
```diff
@@ -3,212 +3,141 @@
...
-                '',
-                '.',
-                '',
-                '',
-            ],
-            static::getSniffClassName()
-        );
+            $this->checkSniffErrors($inputFilePath, $expectedErrors);
+        } else {
+            $this->fail('No fixers nor sniffers were found. Registers them in your config.');
```
[the remark this answers is not in the document]
Comment (alexndlm) [reply tests/Sniffs/AbstractSniffTestCase.php]: Fixed

**Review by ERuban — APPROVED** [review]

**Pull request merged**